backend/app/services/background_chat_executor.py

```
@@ -0,0 +1,392 @@
+            )
+
+            # 4. Call Chat Shell, accumulate complete response
+            accumulated_content = ""
+            async for event in self._adapter.chat(chat_request):
+                if event.type == ChatEventType.CHUNK:
+                    content = event.data.get("content", "")
+                    if content:
+                        accumulated_content += content
+                elif event.type == ChatEventType.ERROR:
+                    error_msg = event.data.get("error", "Unknown error")
+                    raise Exception(error_msg)
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)